src/Symfony/Component/Messenger/Middleware/SendMessageMiddleware.php

```diff
@@ -39,7 +39,11 @@ class SendMessageMiddleware implements MiddlewareInterface
     public function __construct(SendersLocatorInterface $sendersLocator, EventDispatcherInterface $eventDispatcher = null)
     {
         $this->sendersLocator = $sendersLocator;
-        $this->eventDispatcher = LegacyEventDispatcherProxy::decorate($eventDispatcher);
+
+        if (null !== $eventDispatcher) {
```

Comment:
maybe also add a class_exists check?
in master, could replace the contracts by PSR-14 if that makes sense btw

Reply:
I found some other places where we could make use of that too.